### PR TITLE
Add a gradle task for running iOS tests (draft)

### PR DESCRIPTION
**Pull request by saket**
Work in progress.  

Kevin from touchlab was [kind enough](https://twitter.com/kpgalligan/status/1204935729450770433) to update their fork of Koin. That was possibly the final blocker on dependencies to make them play nicely with Kotlin 1.3.60.

In its current state, running `./gradlew build` results in an error. Trying to figure out how to fix this.

```
error: Invalid record (Producer: 'LLVM8.0.0svn' Reader: 'LLVM APPLE_1_1100.0.33.8_0')
1 error generated.
e: Compilation failed: The /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ command returned non-zero exit code: 1.
output:

 * Source files:
 * Compiler version info: Konan: 1.3.60 / Kotlin: 1.3.60
 * Output kind: FRAMEWORK

e: org.jetbrains.kotlin.konan.KonanExternalToolFailure: The /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ command returned non-zero exit code: 1.
```

### CODE REVIEW

**Comment by saket** [comment]
Turns out, [assertk](https://github.com/willowtreeapps/assertk) is also a blocker due to ABI mismatch. I've opened a [PR here](https://github.com/willowtreeapps/opentest4k/pull/9) to start the process of bringing it to 1.3.60.

**Comment by saket** [comment]
Okay, @kpgalligan [helped me](https://github.com/kpgalligan/press/tree/kpg/dec12/no-crash) narrow down the issue: [TakeWhile](https://github.com/saket/press/blob/29927b87c6e6e69fcc153da2696063222f4f07cd/shared/src/commonMain/kotlin/me/saket/press/shared/rx/TakeWhile.kt) was for some reason making the kotlin native compiler explode. The iOS project should now atleast build.

**Comment by saket** [comment]
The tests can now run, but there are a lot of errors that need to be fixed. Will do that in a follow-up PR. This is already a huge win. 🎉

File: shared/src/commonMain/kotlin/me/saket/press/shared/rx/TakeWhile.kt [hunk 1]
```diff
@@ -1,36 +0,0 @@
...
-import com.badoo.reaktive.disposable.DisposableWrapper
-import com.badoo.reaktive.observable.Observable
-import com.badoo.reaktive.observable.ObservableObserver
-import com.badoo.reaktive.observable.observable
-
-/**
- * Emit items by upstream as long as `predicate` is satisfied,
- * and then completes as soon as this condition is not satisfied.
- */
-fun <T> Observable<T>.takeWhile(predicate: (T) -> Boolean): Observable<T> =
```
Comment (Ethan1983) [inline shared/src/commonMain/kotlin/me/saket/press/shared/rx/TakeWhile.kt:15]: Is there a issue request for Kotlin native compiler to support this?

Comment (saket) [reply shared/src/commonMain/kotlin/me/saket/press/shared/rx/TakeWhile.kt:15]: No, I should create one. Thank you for the reminder.

Comment (saket) [reply shared/src/commonMain/kotlin/me/saket/press/shared/rx/TakeWhile.kt:15]: Done: https://youtrack.jetbrains.com/issue/KT-35477

**Pull request merged**